### PR TITLE
Servo example

### PR DESCRIPTION
**Pull request by Kayleb814**
Commented and tested Example code.

### CODE REVIEW

File: INMOOV-Head-Control/Head.cpp [hunk 1]
```diff
@@ -2,98 +2,98 @@
...
+           int servoUpdataRate)                                                //Number of times per second the servos update
+{
+  m_allServos[0] = eyesPanServo;                                              //Add each servo object to the all servos array
+  m_allServos[1] = eyesTiltServo;
+  m_allServos[2] = jawServo;
+  m_allServos[3] = headYawServo;
+  m_allServos[4] = headPitchServo;
+  m_allServos[5] = headRollServo;
+
+  m_servoUpdateRate = servoUpdataRate;                                        //How many times per second the servos' position is updated
```
Comment (gordonso928) [inline INMOOV-Head-Control/Head.cpp]: Maybe changing the "servoUpdateRate" to "servoUpdatePerSecond" will be more intuitive

File: INMOOV-Head-Control/INMOOV-Head-Control.ino [hunk 2]
```diff
@@ -1,7 +1,67 @@
...
+// class, variable object name,
+Head head(servo_1, // objects included in class
+          servo_2, //        "
+          servo_3, //        "
+          servo_4, //        "
+          servo_5, //        "
+          servo_6, //        "
+          40);     //        update frequency
+
+int duration = 500;           //  Time for how long the movement will take place
```
Comment (gordonso928) [inline INMOOV-Head-Control/INMOOV-Head-Control.ino]: Add the units used in the comment

File: INMOOV-Head-Control/INMOOV-Head-Control.ino [hunk 3]
```diff
@@ -1,7 +1,67 @@
...
+volatile bool interrupt1 = 0; //  Variable for the state of the interrupt
```
Comment (gordonso928) [inline INMOOV-Head-Control/INMOOV-Head-Control.ino:22]: Update the variable name to be more descriptive, for example if we are using a hardware interrupt on a button, call the interrupt as "buttonInterruptFlag".

File: INMOOV-Head-Control/INMOOV-Head-Control.ino [hunk 4]
```diff
@@ -1,7 +1,67 @@
...
+  Serial.begin(9600);                  //  Initialize serial comunications
+  randomSeed(analogRead(A0));          //  Use the analog signal from pin A0 to start a random point from
+  pinMode(interruptPin, INPUT_PULLUP); //  Initialize a pin for the interrupt and pull it high
+  /*  Initialize an interrupt pin that throughs an interrupt when it's state changes */
+  attachInterrupt(digitalPinToInterrupt(interruptPin), interrupt, CHANGE);
+}
+
+void loop()
+{
+  /*
```
Comment (gordonso928) [inline INMOOV-Head-Control/INMOOV-Head-Control.ino]: Move the comment out of the loop, maybe like line 50. Also make the comment more easy to read like make it in point form (so the action will be in chunk size)

File: INMOOV-Head-Control/StandardServo.cpp [hunk 5]
```diff
@@ -2,87 +2,87 @@
...
-  }else{
+  } else {
     validMove = true;
   }
 
   target = constrain(target, m_minRotation, m_maxRotation); //Constain target roation to bounds
 
-  if(m_reverseDirection){
+  if (m_reverseDirection) {
     m_servo.writeMicroseconds(int(mapf(target,
```
Comment (gordonso928) [inline INMOOV-Head-Control/StandardServo.cpp:74]: I believe we can just use the normal Arduino map() function, as we will map it to a int anyways. Check if the outcome value will be the same.

**Review by gordonso928 — CHANGES_REQUESTED** [review]
Code looks good! minor changes can make the code better for documentation and formatting purposes.

File: INMOOV-Head-Control/StandardServo.cpp (hunk 5 above)
Comment (sssmc) [reply INMOOV-Head-Control/StandardServo.cpp:74]: The orginal reason for adding mapf as to address a drift issue when using the moveRelative() function. I would say we should leave it in unless we remove the moveRelative function completely.

File: INMOOV-Head-Control/INMOOV-Head-Control.ino (hunk 3 above)
Comment (sssmc) [reply INMOOV-Head-Control/INMOOV-Head-Control.ino:22]: This name was chosen to be generic example placeholder to show how a interrupt is implemented. This intterrupt could be triggered by various hardware or software events and would be named more specifically in final code.

File: INMOOV-Head-Control/INMOOV-Head-Control.ino (hunk 4 above)
Comment (sssmc) [reply INMOOV-Head-Control/INMOOV-Head-Control.ino]: Split into line inline comments

File: INMOOV-Head-Control/INMOOV-Head-Control.ino (hunk 2 above)
Comment (sssmc) [reply INMOOV-Head-Control/INMOOV-Head-Control.ino]: Done

File: INMOOV-Head-Control/Head.cpp (hunk 1 above)
Comment (sssmc) [reply INMOOV-Head-Control/Head.cpp]: Done, change to servoUpdatesPerSecond

**Review by gordonso928 — APPROVED** [review]

**Pull request merged**